lib/resty/cassandra/cluster.lua

```diff
@@ -550,6 +551,15 @@ function _Cluster:refresh()
   -- initiate the load balancing policy
   self.lb_policy:init(peers)
 
+  if self.stream_ids == nil then
+    -- Initialize the list of available seed ids (only once)
+    self.stream_ids = {}
```

Comment:
True, I could use a counter and re-initialize it when we get over `2^7`, however that gives us less features than this implementation.

Having a list of available ids and removing and reinserting them means only non currently used ids will be available to get. With a counter, there's no way to keep track of which ids are still being used. This might not be a big deal in the protocol v3 where we can use 32767 ids, but might be an issue with protocol v2 since we only have 127 available ids there.

The list of ids also gives us the advantage that we'll always be using the least recently used (released) id since they're added and removed in order.

This logic is similar to what's done in the python datastax driver: https://github.com/datastax/python-driver/blob/6bec6fd7e852ae33e12cf475b030d08260b04240/cassandra/connection.py#L286